js/classes/asset_manager.js

```diff
@@ -75,5 +113,64 @@ class AssetManager {
     clearCache() {
         this.cache = [];
     };
+
+    /**
+     * This method plays the audio associated with the given path.
+     * @param {string} path The filepath of the audio you are trying to play.
+     * @param {number} volume The volume to which you want to set the audio.
+     * @param {boolean} loop True if you want the audio to loop, false otherwise.
+     */
+    playAudio(path, volume, loop) {
+        loop ?? (loop = false);
+        const audio = this.cache[path];
+        audio.currentTime = 0;
+        audio.volume = volume;
+
+        audio.play();
+        if (loop) {
+            audio.addEventListener("ended", () => {
+                audio.play();
+            });
+        }
+    };
+
+    /**
+     * This method pauses the audio associated with the given path.
+     * @param {boolean} mute True if you want to mute the audio, false otherwise.
+     */
+    muteAudio(mute) {
+        for (let key in this.cache) {
+            const audio = this.cache[key];
+            if (audio instanceof Audio) {
+                audio.muted = mute;
+            }
+        }
+    };
+
+    /** 
+     * This method sets the volume of all audio in the cache to the given volume.
+     * @param {number} volume The volume to which you want to set the audio.
+     */
+    adjustVolume(volume) {
+        for (let key in this.cache) {
```

Comment:
If i was doing this for each loop, i would do it a lil different.

this.cache.forEach( ( key ) => {
    const audio = this.cache[key];
    ..........
});

If yours works, whatever. I've just never seen a for each loop like yours before.

Reply:
Good idea! I made the change.